server/src/main/java/org/elasticsearch/index/query/IntervalsSourceProvider.java

```diff
@@ -147,7 +147,7 @@ public IntervalsSource getSource(QueryShardContext context, MappedFieldType fiel
             }
             IntervalsSource source;
             if (useField != null) {
-                fieldType = context.fieldMapper(useField);
+                fieldType = context.getFieldType(useField);
                 assert fieldType != null;
```

Comment:
These assertions look dodgy (I think I put them in in the first place, so I'm allowed to say that).  It should be throwing an error if passed a field that doesn't exist or doesn't support text searches.